rclcpp/src/rclcpp/generic_publisher.cpp

```diff
@@ -31,4 +31,49 @@ void GenericPublisher::publish(const rclcpp::SerializedMessage & message)
   }
 }
 
+void GenericPublisher::publish_loaned_msg(const rclcpp::SerializedMessage & message)
```

Comment:
Since this isn't actually publishing a loaned message, instead its publishing a given message **as** as loaned message internally, I think this API would need to be `publish_msg_as_loaned` or `publish_as_loaned_msg` to make the distinction that this function does not publish a pre-borrowed message its being given as input, but transfers a given message into the loaned set of memory locally.

Reply:
Yes. I will update code to use publish_as_loaned_msg.